apps/wolfsshd/configuration.c

```diff
@@ -490,69 +496,114 @@ static int HandleInclude(WOLFSSHD_CONFIG *conf, const char *value)
+                            WMEMMOVE(fileNames+j+1, fileNames+j,
+                                    (i - j)*sizeof(char*));
+                            break;
                         }
                     }
-                    if (postfix) {
-                        /* Skip if file is too short */
-                        if (WSTRLEN(dir->d_name) <= WSTRLEN(postfix)) {
-                            continue;
-                        }
-                        if (WSTRNCMP(dir->d_name + WSTRLEN(dir->d_name) -
-                                    WSTRLEN(postfix), postfix, WSTRLEN(postfix))
-                                == 0) {
-                            snprintf(filepath, PATH_MAX, "%s/%s", path,
-                                     dir->d_name);
-                        }
-                        else {
-                            /* Not a match */
-                            continue;
-                        }
+                    fileNames[j] = dir->d_name;
+                    i++;
+                }
+            }
+
+            for (i = 0; i < fileCount; i++) {
+                /* Check if filename prefix matches */
+                if (prefixLen > 0) {
+                    if ((int)WSTRLEN(fileNames[i]) <= prefixLen) {
+                        continue;
                     }
-                    else {
-                        snprintf(filepath, PATH_MAX, "%s/%s", path,
-                                 dir->d_name);
+                    if (WSTRNCMP(fileNames[i], prefix, prefixLen) != 0) {
+                        continue;
                     }
-                    ret = wolfSSHD_ConfigLoad(conf, filepath);
-                    if (ret != WS_SUCCESS) {
-                        return ret;
+                }
+                if (postfix) {
+                    /* Skip if file is too short */
+                    if (WSTRLEN(fileNames[i]) <= WSTRLEN(postfix)) {
+                        continue;
                     }
+                    if (WSTRNCMP(fileNames[i] + WSTRLEN(fileNames[i]) -
+                                WSTRLEN(postfix), postfix, WSTRLEN(postfix))
+                            == 0) {
+                        WSNPRINTF(filepath, PATH_MAX, "%s/%s", path,
+                                 fileNames[i]);
+                    }
+                    else {
+                        /* Not a match */
+                        continue;
+                    }
+                }
+                else {
+                    WSNPRINTF(filepath, PATH_MAX, "%s/%s", path,
+                             fileNames[i]);
+                }
+                ret = wolfSSHD_ConfigLoad(conf, filepath);
+                if (ret != WS_SUCCESS) {
+                    WCLOSEDIR(&d);
+                    WFREE(fileNames, conf->heap, DYNTYPE_PATH);
+                    WFREE(filepath, conf->heap, DYNTYPE_PATH);
+                    WFREE(path, conf->heap, DYNTYPE_PATH);
+                    return ret;
```

Comment:
Would prefer this function have a single return at the end, which'll save us having to do all the freeing at multiple different return sites, but not critical.